### PR TITLE
feat: if profile only has one public event-type, redirect to it 

### LINKED ISSUE #12089: [CAL-2648] if profile only has one public event-type, redirect to it
1. https://cal.com/peer only has one visible event-type (others are secret/hidden)
2. visiting https://cal.com/peer should redirect to https://cal.com/peer/meet immediately
3. however, if you click the avatar, it should go back to profile without redirecting (i.e. as a URL parameter: ?redirect=false)
CleanShot 2023-10-25 at 20 36 10@2x

# why:
- having only one event-type means a user can only take one action anyway
- speeds up, so people can send cal.com/peer vs cal.com/peer/meet

<sub>[CAL-2648](https://linear.app/calcom/issue/CAL-2648/if-profile-only-has-one-public-event-type-redirect-to-it)</sub>

### PR DESCRIPTION
**Pull request by Dhoni77**
## What does this PR do?

- Redirect user directly to the event if only a single public event is present
- When clicked on the avatar in the event, navigate to the user profile

Fixes #12089

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- User must have a single public event type.
- Navigate to https://cal.com/user_name
- The page should be redirected to that public event
- When clicked on the avatar it should navigate to the user profile

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/components/ui/avatar/UserAvatarGroup.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export function UserAvatarGroup(props: UserAvatarProps) {
         alt: user.name || "",
         title: user.name || "",
         image: getUserAvatarUrl(user),
+        href: "/settings/my-account/profile",
```
Comment (PeerRich) [inline apps/web/components/ui/avatar/UserAvatarGroup.tsx:17]:
```suggestion
```

**Comment by PeerRich** [comment]
does this also work with secret events?

File: apps/web/pages/[user].tsx [hunk 2]
```diff
@@ -380,6 +380,20 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
...
+    return {
+      redirect: {
+        permanent: false,
+        destination: `${user.username}/${eventTypesRaw[0].slug}`,
+      },
+    } as {
+      redirect: {
+        permanent: false;
+        destination: string;
+      };
```
Comment (PeerRich) [inline apps/web/pages/[user].tsx:393]: what is the "as" statement for?

**Comment by PeerRich** [comment]
this is not fully implemented correctly. the idea is to be able to visit the profile page if you click the avatar, skipping the redirect

**Review by Udit-takkar — COMMENTED** [review]
@Dhoni77 any update on this ?

You have not implemented the 3rd point mentioned in the issue
?redirect=false

**Comment by Dhoni77** [comment]
> @Dhoni77 any update on this ?
> 
> You have not implemented the 3rd point mentioned in the issue ?redirect=false

Hi @Udit-takkar I can see there is a pr that fixes this issue and is approved. So I think you can close this.
Thanks

**Pull request closed without merging**